hpx/runtime/serialization/array.hpp

```diff
@@ -99,6 +98,13 @@ namespace hpx { namespace serialization
         return array<T>(begin, size);
     }
 
+    // implement serialization for arrays
+    template <class Archive, class T, std::size_t N>
+    void serialize(Archive& ar, T (&a)[N], const unsigned int /* version */)
+    {
+        ar & hpx::serialization::make_array(a, N);
+    }
+
     // implement serialization for boost::array
```

Comment:
I don't know. I did not make any changes to apply/async. All I care for is that I can now serialize objects a la

struct foo {
  int bar[40];
};